tools/codegen/gen.py

```diff
@@ -1069,6 +1069,11 @@ def gen_source_files(
 #include <ATen/hip/HIPDevice.h>
 #include <ATen/hip/HIPContext.h>'''
 
+    # Only the set of keys defined in dispatch_keys should be supported,
+    # raise an error for unsupported keys
+    for k in backend_indices.keys():
+        assert k in dispatch_keys, f"{k} is not a supported dispatch key."
```

Comment:
This doesn't seem quite right. For one, there is a `--backend-whitelist` which optionally controls what is in `dispatch_keys`, but we ought not fail if `native_functions.yaml` has a backend that wasn't in the whitelist. I'd probably put the check here in model.py and maintain a non-wobbling list of all valid dispatch keys for codegen, e.g., factor out this chunk:

```
    dispatch_keys = [
        DispatchKey.CPU,
        DispatchKey.SparseCPU,
        DispatchKey.SparseCsrCPU,
        DispatchKey.MkldnnCPU,
        DispatchKey.CUDA,
        DispatchKey.SparseCUDA,
        DispatchKey.SparseCsrCUDA,
        DispatchKey.QuantizedCPU,
        DispatchKey.QuantizedCUDA,
        DispatchKey.CompositeImplicitAutograd,
        DispatchKey.CompositeExplicitAutograd,
        # Meta is a magic key: it is automatically generated for structured
        # kernels
        DispatchKey.Meta,
        DispatchKey.ZeroTensor,
    ]
```

Reply:
yeah, this is why your android builds are failing